### PR TITLE
Add a version package, populate it and use it

### PR DESCRIPTION
**Pull request by stevekuznetsov**
The last commit was generated with:

```py
#!/usr/bin/env python

import sys

out = []
imageIndex = 0
goBinaryIndex = 0
declaration = ""
with open(sys.argv[1]) as f:
	for i, line in enumerate(f):
		out.append(line)
		if "prow_image(" in line:
			imageIndex = i
			out.append("    component = NAME,\n")

		if "go_binary(" in line:
			goBinaryIndex = i

		if i == goBinaryIndex + 1 and "name" in line:
			# ex. "    name = "deck","
			component = line.split("\"")[1]
			out = out[:-1] # undo the name declaration in go_binary
			declaration = "NAME = \"{}\"\n".format(component) # declare the shared var
			out.append("    name = NAME,\n") # rewrite the name declaration to use it

		if imageIndex != 0 and declaration != "":
			# we need to insert at the earliest declaration that will use it
			insertIndex = imageIndex
			if goBinaryIndex < insertIndex:
				insertIndex = goBinaryIndex
			out.insert(insertIndex, declaration)
			declaration = ""

with open(sys.argv[1], 'w') as f:
    for line in out:
    	f.write(line)


# manually edit:
# experiment/resultstore/BUILD.bazel (prow_image was not indented)
# prow/cmd/peribolos/BUILD.bazel (e2e images don't need rename)
# robots/pr-labeler/BUILD.bazel (prow_image was not indented)
```

/assign @ixdy @fejta @cjwagner @alvaroaleman

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Had to manually edit coinfigurator, too, it had an issue

**Comment by stevekuznetsov** [comment]
Good lord, is everything formatted now? All the boilerplate fine? ew.

**Comment by stevekuznetsov** [comment]
```
FAIL: @io_k8s_repo_infra//hack:verify-boilerplate (see /bazel-scratch/.cache/bazel/_bazel_root/05618a594cb3499a4b912817865867cd/execroot/io_k8s_test_infra/bazel-out/k8-fastbuild/testlogs/external/io_k8s_repo_infra/hack/verify-boilerplate/test.log)
INFO: From Testing @io_k8s_repo_infra//hack:verify-boilerplate:
==================== Test output for @io_k8s_repo_infra//hack:verify-boilerplate:
1 files have incorrect boilerplate headers:
prow/version/doc.go 
```

Why?!

**Comment by stevekuznetsov** [comment]
Ah, I had a trailing space ...

**Comment by stevekuznetsov** [comment]
Finally!

File: prow/version/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,23 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["doc.go"],
+    importpath = "k8s.io/test-infra/prow/version",
+    visibility = ["//visibility:public"],
+    x_defs = {"Version": "{DOCKER_TAG}"},
```
[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply prow/version/BUILD.bazel]: IIRC this worked as expected last time we did this, so I did not look.

**Comment by stevekuznetsov** [comment]
Yes, this works as expected. 

Locally, I changed `clonerefs` to do this:

```go
logrus.Info(version.UserAgent())
os.Exit(1)
```

Then, I run it:

```
$ bazel run //prow/cmd/clonerefs:image
// ...
Tagging 4bb667121caa382eed5f2c29b53fa5476fad62771ae33cef971acf418bba6eab as bazel/prow/cmd/clonerefs:image
$ docker run 4bb667121caa382eed5f2c29b53fa5476fad62771ae33cef971acf418bba6eab
{"component":"clonerefs","file":"prow/cmd/clonerefs/main.go:31","func":"main.main","level":"info","msg":"clonerefs/v20200313-0d4dcac83-dirty","time":"2020-03-13T23:27:45Z"}
```

So the `component` field in the log and the user agent are set as expected

File: prow/version/BUILD.bazel (hunk 1 above)
[the remark this answers is not in the document]
Comment (ixdy) [reply prow/version/BUILD.bazel]: yes: https://github.com/bazelbuild/rules_go/blob/master/go/core.rst#stamping-with-the-workspace-status-script

**Pull request merged**